src/effects/effectparameter.h

```diff
@@ -5,59 +5,27 @@
 #include <QVariant>
 
 #include "effects/effectmanifestparameter.h"
+#include "effects/effectslot.h"
 #include "util/class.h"
 
 class Effect;
 class EffectsManager;
 
-// An EffectParameter is an instance of an EffectManifestParameter, which is in
-// charge of keeping track of the instance values for the default, minimum,
-// maximum and value for each Effect's parameter, and validating that they are
-// always within acceptable ranges. This class is NOT thread-safe and must only
-// be used from the main thread.
+// An EffectParameter is a wrapper around EffectManifestParameter that tracks a
+// mutable value state and communicates that state to the engine. This class is
+// NOT thread-safe and must only be used from the main thread.
 class EffectParameter : public QObject {
     Q_OBJECT
   public:
-    EffectParameter(Effect* pEffect, EffectsManager* pEffectsManager,
+    EffectParameter(EffectSlot* pEffectSlot, EffectsManager* pEffectsManager,
                     int iParameterNumber, EffectManifestParameterPointer pParameter);
     virtual ~EffectParameter();
 
-    void addToEngine();
-    void removeFromEngine();
-
-    ///////////////////////////////////////////////////////////////////////////
-    // Parameter Information
-    ///////////////////////////////////////////////////////////////////////////
-
     EffectManifestParameterPointer manifest() const;
```

Comment:
rename to `getParameterManifest` to avoid potential confusion with EffectManifest